atlp-rwanda/atlp-pulse-mobile
# #2 Implement user registration flow

### What does this PR do?

This PR includes design and implementation of trainee registration.

### Description of Task to be completed?

Once a trainee is invited in an organization, they receive an email with a link to register. Once they click on the link they will be redirected to the app on the registration page. The page receives a token and a user fills in their information. Once they submit, an account is created and they are redirected to the trainee dashboard.

### How should this be manually tested?

- Invite a user using the frontend and make sure they receive an invitation email (set smtp related environment variables to be able to send emails).
- On the testing device, open the invitation email and click on it, the app will be opened on the registration page.
- Fill in all required information and submit the form.

### Screenshots
<img width="300" src="https://github.com/user-attachments/assets/67acfc9a-6580-4dd7-b906-29c641453688" />
<img width="300" src="https://github.com/user-attachments/assets/133c3a11-c7f5-4411-98a2-921c153b80c0" />
